### PR TITLE
Login and signup

### PR DESCRIPTION
**Pull request by abaddonpuff**
Added login, logout, signup and installed pre-commit. Improved readibnility and cleaned up some unused files. Ready to push to heroku once approved :)

### CODE REVIEW

**Comment by abaddonpuff** [comment]
I do have an error I cannot fix right now. When I log in everything works when authenticated. EXCEPT when I tried to read posts.... we can speak about it on Thursday, but when I try to read the posts on any of the X users, it redirects to login page as if it was not authenticated. Everything else works great.

File: centralHub/migrations/0004_twitteruser_twitter_name.py [hunk 1]
```diff
@@ -2,22 +2,23 @@
 
 from django.db import migrations, models
 
-def populate_unique_emails(apps, schema_editor):
```
Comment (bbelderbos) [inline centralHub/migrations/0004_twitteruser_twitter_name.py:5]: What was this for?

**Review by bbelderbos — COMMENTED** [review]
Seeing mostly formatting, let me try it out and verify later ...

File: centralHub/migrations/0004_twitteruser_twitter_name.py (hunk 1 above)
Comment (abaddonpuff) [reply centralHub/migrations/0004_twitteruser_twitter_name.py:5]: It was a script I used to fill a schema change.

**Comment by abaddonpuff** [comment]
> Seeing mostly formatting, let me try it out and verify later ...

Yeah, I run ruff on the entire folder too, sorry for that. Changes are in the accounts folder (almost root) which handles signup. On Settings for the main application and on the base template and login/logout/signup forms. I added is.authenticated in the base and then extend base to the rest. But it is only the X Posts that have errors, the Spotify artist, albums and the X users work

**Pull request merged**